strawberryfields/api/job.py

```diff
@@ -119,14 +121,27 @@ def result(self) -> Result:
             )
         return self._result
 
+    @property
+    def meta(self) -> dict:
+        """Returns a dictionary of metadata on job execution, such as error
+        details.
+
+        Returns:
+            dict[str, str]
+        """
+        return self._meta
+
     def refresh(self):
-        """Refreshes the status of an open or queued job,
+        """Refreshes the status and metadata of an open or queued job,
         along with the job result if the job is newly completed.
         """
         if self._status.is_final:
             self.log.warning("A %s job cannot be refreshed", self._status.value)
             return
-        self._status = JobStatus(self._connection.get_job_status(self.id))
+        job_info = self._connection.get_job(self.id)
+        self._status = JobStatus(job_info.status)
+        self._meta = job_info.meta
+        self.log.debug("Job {} metadata: {}".format(self.id, job_info.meta))
```

Comment:
Thanks for adding this! Usually using the old string formatting syntax is suggested for logging, I also looked up just now [a couple of further reasons to it](https://reinout.vanrees.org/weblog/2015/06/05/logging-formatting.html). 
```suggestion
        self.log.debug("Job %s metadata: %s", self.id, job_info.meta)
```

Reply:
Good catch, thanks!